src/style/form/form.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useState } from 'react';
+import './style-form.css';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faExclamationTriangle } from '@fortawesome/free-solid-svg-icons';
+import { IconDefinition } from '@fortawesome/free-solid-svg-icons';
+import { colors } from '../colors';
+import { Label } from '../typography/LargeLabel';
+import { Caption } from '../typography/caption';
+
+interface FormProps {
+  caption?: string;
+  label?: string;
+  iconError?: IconDefinition;
+  icon?: IconDefinition;
+  password?: boolean;
+}
+
+export const FormField = ({ label, icon, password, caption }: FormProps) => {
+  const [focused, setFocused] = useState(false);
+  const [inputValue, setInputValue] = useState('');
+  const [error, setError] = useState(false);
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value;
+    setInputValue(value);
+    if (value === '') {
+      setError(true);
+    } else {
+      setError(false);
+    }
+  };
```

Comment:
Essa lógica não está errada, porém a estilização de erro deveria estar acompanhada do caption, porque sempre que houver uma validação do campo e ele estiver errado, deveria-se mostrar a mensagem de erro em baixo.

Outro ponto é que não é apenas o campo estar ou não preenchido que denota erro. O erro depende do propósito do campo. Por exemplo, se for um campo de e-mail obrigatório, pode acontecer erro qundo ele não está preenchido, mas também quando o usuário digita algo que não é um e-mail. Então essa parte da validação a gente vai deixar pra fazer mais pro futuro, ok?

Como o caption está sendo controlado de fora (via props), provavelmente vc pode fazer o controle dos erros por fora tbm, o que acha?